### PR TITLE
Implement verifySingleCertificateSignature - Closes #6841

### LINKED ISSUE #6841: Implement verifySingleCertificateSignature function of Certificate
### Description

Add logic to `verifySingleCertificateSignature` to Certificate class according to [LIP:Certificate:verifySingleCertificateSignature](https://research.lisk.com/t/introduce-certificate-generation-mechanism/296#verifysinglecertificatesignature-22)

### Acceptance Criteria

- Add unit test for verifySingleCertificateSignature

### PR DESCRIPTION
**Pull request by mehmetegemen**
### What was the problem?

This PR resolves #6841 

### How was it solved?

- Implemented util function

### How was it tested?

- Created unit tests

### CODE REVIEW

File: framework/test/unit/node/consensus/certificate_generation/utils.spec.ts [hunk 1]
```diff
@@ -12,16 +12,94 @@
...
+				signature,
+				networkIdentifier,
+				certificate,
+			);
+
+			expect(isVerifiedSignature).toBeTrue();
+		});
+
+		it('should return false for wrong public key', () => {
+			publicKey = getRandomBytes(32);
```
Comment (shuse2) [inline framework/test/unit/node/consensus/certificate_generation/utils.spec.ts:79]: BLS  public key is 48 bytes

File: framework/test/unit/node/consensus/certificate_generation/utils.spec.ts [hunk 2]
```diff
@@ -12,16 +12,94 @@
...
+
+			signature = signBLS(
+				MESSAGE_TAG_CERTIFICATE,
+				networkIdentifier,
+				encodedCertificate,
+				privateKey,
+			);
+
+			(certificate as any).aggregationBits = Buffer.alloc(0);
+			(certificate as any).signature = Buffer.alloc(0);
```
Comment (shuse2) [inline framework/test/unit/node/consensus/certificate_generation/utils.spec.ts:64]: maybe better to assign actual  bytes to see  if it's ignored

File: framework/test/unit/node/consensus/certificate_generation/utils.spec.ts (hunk 1 above)
Comment (mehmetegemen) [reply framework/test/unit/node/consensus/certificate_generation/utils.spec.ts:79]: done

File: framework/test/unit/node/consensus/certificate_generation/utils.spec.ts (hunk 2 above)
Comment (mehmetegemen) [reply framework/test/unit/node/consensus/certificate_generation/utils.spec.ts:64]: done

**Review by shuse2 — APPROVED** [review]

**Review by mitsuaki-u — APPROVED** [review]

**Pull request merged**